modules/tinymce/src/core/test/ts/browser/keyboard/AutocompleterTest.ts

```diff
@@ -9,64 +10,92 @@ import { EditorEvent } from 'tinymce/core/api/PublicApi';
 import { AutocompleterEventArgs, AutocompleteLookupData } from 'tinymce/core/autocomplete/AutocompleteTypes';
 
 describe('browser.tinymce.core.keyboard.AutocompleterTest', () => {
-  const triggerChar = '+';
+  const plusTriggerChar = '+';
+  const dollarsTriggerChar = '$';
+  // This matches the throttle time used in Autocompleter.ts
+  const keyboardThrottleTimer = 50;
+
   const hook = TinyHooks.bddSetupLight<Editor>({
     base_url: '/project/tinymce/js/tinymce',
     setup: (ed: Editor) => {
+      const onAction = (autocompleteApi: InlineContent.AutocompleterInstanceApi, rng: Range, value: string) => {
+        ed.selection.setRng(rng);
+        ed.insertContent(value);
+        autocompleteApi.hide();
+      };
+
+      const fetch = (ch: string, type: string) => (resolve: (data: InlineContent.AutocompleterContents[]) => void) => {
+        resolve(Arr.map([ 'aa', 'ab' ], (letter) => ({
+          value: `${type}-${letter}`,
+          text: `p-${letter}`,
+          icon: ch
+        })));
+      };
+
       ed.ui.registry.addAutocompleter('Plus1', {
-        ch: triggerChar,
+        ch: plusTriggerChar,
+        minChars: 0,
+        columns: 1,
+        fetch: (_pattern, _maxResults) => new Promise(fetch(plusTriggerChar, 'plus')),
+        onAction
+      });
+
+      const dollarsFetch = Throttler.last(fetch(dollarsTriggerChar, 'dollars'), keyboardThrottleTimer * 3);
+      ed.ui.registry.addAutocompleter('Dollars1', {
+        ch: dollarsTriggerChar,
         minChars: 0,
         columns: 1,
-        fetch: (_pattern, _maxResults) => new Promise((resolve) => {
-          resolve(
-            Arr.map([ 'aa', 'ab' ], (letter) => ({
-              value: `plus-${letter}`,
-              text: `p-${letter}`,
-              icon: '+'
-            }))
-          );
-        }),
-        onAction: (autocompleteApi, rng, value) => {
-          ed.selection.setRng(rng);
-          ed.insertContent(value);
-          autocompleteApi.hide();
-        }
+        fetch: (_pattern, _maxResults) => new Promise(dollarsFetch.throttle),
+        onAction
       });
     }
   }, []);
 
-  const pOpenAutocompleter = async (editor: Editor) => {
+  const pOpenAutocompleter = async (editor: Editor, triggerChar: string) => {
     editor.focus();
     editor.setContent(`<p>${triggerChar}</p>`);
-    TinySelections.setCursor(editor, [ 0, 0 ], triggerChar.length);
-    TinyContentActions.keypress(editor, triggerChar.charCodeAt(0));
+    TinySelections.setCursor(editor, [ 0, 0 ], plusTriggerChar.length);
```

Comment:
Should this be `tiggerChar` instead and for below as well?
```suggestion
    TinySelections.setCursor(editor, [ 0, 0 ], triggerChar.length);
```

Reply:
Yes it should be thanks!